nemo/collections/nlp/modules/common/megatron/layer_norm_1p.py

```diff
@@ -47,3 +48,36 @@ def forward(self, x):
     class LayerNorm1P(nn.Module):
         def __init__(self, *args, **kwargs):
             raise NotImplementedError('LayerNorm1P available only with apex installed')
+
+
+def _cast_if_autocast_enabled(tensor):
```

Comment:
Why do we need this?

Reply:
I added this helper method in order to ensure the correct autocast type for CPU as well as GPU. I noticed that Nemo NLP seems to use the apex library method `_cast_if_autocast_enabled`, however, that method only autocasts correctly when on GPU, not on CPU, as it has no code pathway for accessing `torch.get_autocast_cpu_dtype()`. You can see that here: https://github.com/NVIDIA/apex/blob/05091d498d21058a0fe736b828c43431d4f0dda2/apex/_autocast_utils.py#L22

To ensure we get 1:1 equivalence for MPT even on CPU, I therefore added this to ensure CPU doesn't cast to float32 which is wasteful, and wouldn't make it a low precision layernorm.

I can remove the helper method though and switch to the apex version if you like.